woocommerce/woocommerce-ios
# StorePickerError: Landscape & Big Fonts support

closes #4383 

# Why 

After #5006 is merged, the final bits to tackle are to properly support landscape and accessibility fonts.

This however turned to not be simple. Thanks `SwiftUI` :rage4: 

# How

## Landscape

The first issue is that apparently a `HostingViewController` presented with a `.custom` `modalPresentationStyle` does not inherit rotation support. 

After inspecting the view with the debugger, I noticed that the hosting controller view frame wasn't being updated properly.

To fix, this I'm creating a subclass of `FancyAlertPresentationController` where I manually update the `presentedView` in the `containerViewWillLayoutSubviews` method.

## Scroll & Big Fonts

Normally, in UIKIt, we would add a scroll view as a parent view of the stack view to make the content scroll if it overflows the scroll view size. This was not so trivial in `SwiftUI`. Adding a scroll view crates two problems: A. The scroll view takes all the available size, making the stack view content to be rendered at the top and not centered and B. There is no way to prevent the scroll view to scroll when the view is small to not need from a scroll.

To solve that, I've created a `ViewModifier` that wraps the content inside a scroll view if the view needs to scroll(this is achieved by using a couple of geometry readers.

Note: This view modifier only works for vertical scroll but it should be easy to adapt it to handle horizontal scroll.

# Demo

## Landscape

https://user-images.githubusercontent.com/562080/133705412-1b23c88f-03bf-4504-a4c2-54fc61bf0b84.mov

## Big Fonts

https://user-images.githubusercontent.com/562080/133705465-d6d7a231-cd1c-499a-bdf5-3eee04648370.mov

# Testing Steps

- Navigate to the store picker screen
- Turn off wifi in order to simulate a network error
- Select another store
- See error appearing as a modal.
- See that you can rotate it and scroll all the content.


Update release notes:
